app/controllers/application_controller.rb

```diff
@@ -1,2 +1,21 @@
 class ApplicationController < ActionController::Base
+    add_flash_types  :success
+    
+    helper_method :current_user, :logged_in?
+
+    def current_user
+        @current_user ||= User.find_by_id(session[:user_id])
+
+    end
+    
+    def logged_in?
+        current_user != nil
```

Comment:
Not something you have to change but a nice trick you can use is called a double BANG!, a bang is `!`.
So here you want to see if `current_user` is not false. If `current_user` was `nil`, then `!current_user` would be true, what we can do is double BANG `!!current_user` so this means `nil` or `undefined` would return `false` and if `current_user` exists it will return `true`

Reply:
The same as a double negative, 'Current user does not not exist'.

Reply:
(optional)